### ISSUE TITLE
[Feature request] Add auth_kubernetes_* resources

### ISSUE THREAD

**Issue by syndbg** [terraform-providers/terraform-provider-vault#93]
# Motivation

tl;dr Add auth_kubernetes_config and similar resources 

Long:

I've previously setup Vault authentication using Kubernetes via CLI, but as a means of automation,
at work we're considering usage of Terraform to do that instead of bash scripts left and right.

We already use Terraform heavily for server provisioning wherever applicable, so why not provision Vault too?

With the most recent release `v1.1.0`, the addition of `approle_auth_backend_*` resources, seems like a step towards adding authentication-provider specific resources wherever applicable.

Currently setting up Vault authentication with Kubernetes with `vault_generic_secret` resources is not possible. From what I've read, there are issues such as https://github.com/hashicorp/vault-plugin-auth-kubernetes/issues/13 and https://github.com/terraform-providers/terraform-provider-vault/issues/44.

From my experience with Vault setup so far - I feel like using the wrong tools as `vault_generic_secret` to do the job. 

### Terraform Version

`Terraform v0.11.7`
 

### Affected Resource(s)

No existing resources, yet.

### References

- GH-93 (actual PR)

**Comment by syndbg** [comment]
Closing since https://github.com/terraform-providers/terraform-provider-vault/pull/94 is merged
